src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java

```
@@ -0,0 +1,386 @@
+            case MQFB_COA:
+                return MQReportConstants.MQFB_COA;
+            case MQFB_COD:
+                return MQReportConstants.MQFB_COD;
+            case MQFB_QUIT:
+                return MQReportConstants.MQFB_QUIT;
+            case MQFB_APPL_CANNOT_BE_STARTED:
+                return MQReportConstants.MQFB_APPL_CANNOT_BE_STARTED;
+            case MQFB_TM_ERROR:
+                return MQReportConstants.MQFB_TM_ERROR;
+            case MQFB_APPL_TYPE_ERROR:
+                return MQReportConstants.MQFB_APPL_TYPE_ERROR;
+            case MQFB_STOPPED_BY_MSG_EXIT:
+                return MQReportConstants.MQFB_STOPPED_BY_MSG_EXIT;
+            case MQFB_XMIT_Q_MSG_ERROR:
+                return MQReportConstants.MQFB_XMIT_Q_MSG_ERROR;
+            case MQFB_ACTIVITY:
+                return MQReportConstants.MQFB_ACTIVITY;
+            case MQFB_MAX_ACTIVITIES:
+                return MQReportConstants.MQFB_MAX_ACTIVITIES;
+            case MQFB_NOT_FORWARDED:
+                return MQReportConstants.MQFB_NOT_FORWARDED;
+            case MQFB_NOT_DELIVERED:
+                return MQReportConstants.MQFB_NOT_DELIVERED;
+            case MQFB_UNSUPPORTED_FORWARDING:
+                return MQReportConstants.MQFB_UNSUPPORTED_FORWARDING;
+            case MQFB_UNSUPPORTED_DELIVERY:
+                return MQReportConstants.MQFB_UNSUPPORTED_DELIVERY;
+            default:
+                return MQReportConstants.MQFB_NONE;
+        }
+    }
+
+    /**
+     * Initialize reply queue
+     */
+    MQQueue setReplyQueue(MQConnectionBuilder connectionBuilder, MQConfiguration config) {
+        MQQueueManager manager = connectionBuilder.getManager();
+        MQQueue queue;
+        try {
+            queue = manager.accessQueue(config.getreplyQueue(), CMQC.MQRC_READ_AHEAD_MSGS);
+        } catch (MQException e) {
+            log.error("Error creating reply queue" + e);
+            return null;
```

Returning null and handle error as generic, we will be loosing information why this got failed. It s better if we could throw this exception to higher layer and handle over there.